### DISCUSSION TITLE
About WidgetHandle

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#185] [General]
Sometimes I get to the situation where I'm using the same capsule multiple times with the same `WidgetHandle` instance. In my case is the `languageCapsule` to do some translations.

Is this bad? Should I pass down the value via function parameter instead using it multiple times? Will it result in multiple rebuilds when the readed capsule rebuilds?

**Comment by busslina** [comment]
> Will it result in multiple rebuilds when the readed capsule rebuilds?

I understand now that no, because `dirty` widget logic is not sync.

**Comment by GregoryConrad** [comment]
If you mean:

```dart
@rearchWidget
Widget myWidget(WidgetHandle use) {
  use(aCapsule);
  use(aCapsule);
  use(aCapsule);
  return Placeholder();
}
```

Then that's fine/supported.

**Reply by busslina** [reply]
Yeah, similar, on subfunctions. Thanks
